src/main/java/com/hillel/items_exchange/service/impl/AdvertisementServiceImpl.java

```diff
@@ -149,6 +157,19 @@ public void setDefaultImage(Advertisement advertisement, Long imageId, User owne
         }.getType());
     }
 
+    private List<AdvertisementTitleDto> mapAdvertisementsToTitleDto(Collection<Advertisement> advertisements) {
+        return advertisements.stream().map(advertisement ->
+                AdvertisementTitleDto.builder()
```

Comment:
You should extract builder into another method **buildAdvertisementTitle** or something similar to and use is here with link to the method ::

Reply:
Or you could use **  BeanUtils.copyProperties** but you all the fields names should be the same

Reply:
Good point. Thnx.